### PR TITLE
Only return active domains for processing calculated properties

### PR DESCRIPTION
**Pull request by gherceg**
## Technical Summary

Came up because of work I was doing in https://dimagi-dev.atlassian.net/browse/SAAS-14915, but is a bit of a side quest.

If a domain is not active, there is no need to update calculated properties for it. This PR updates the filter used for obtaining domains to update calculated properties for.

I first wrote tests for the existing code, and then added the active domain filter with a new test. In my last commit (4176923c42132922f0b0afdfe7606f24ff0571ea), I removed the filter that only looked for forms submitted in the last day if the domain had been updated in the last week. I don't think this optimizes the query or anything, and therefore do not think it is necessary. Please correct me if I'm wrong 👍🏻 

Review by commit 🐡 

### Safety story

Calculated properties are used internally, but that doesn't excuse any issues with this code. With the addition of tests, I feel better about this code now than previously.

### Automated test coverage

Added tests for the method being updated.

### QA Plan

No QA

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

**Comment by gherceg** [comment]
I posed a question in my [most recent update](https://dimagi-dev.atlassian.net/browse/SAAS-14915?focusedCommentId=286919) on the ticket
> In the worst case, if forms were submitted the day the domain was deactivated, the calculated properties would not reflect that. The domain filter checks for domains with submissions in the last day already, so we could use this info to make sure a domain is both deactivated and doesn’t have submissions from within the last day before excluding it.

I'm tempted to update the PR to still run calculated properties on inactive domains with submissions in the last day (or the most robust would be submissions since the last time calculated properties were updated). I'll look into confirming that today.

**Review by millerdev — APPROVED** [review]

File: corehq/apps/reports/tests/test_tasks.py [hunk 1]
```diff
@@ -65,6 +75,79 @@ def test_returns_none_when_found_as_rel_path_basename(self):
...
+        domain = self.index_domain('form-from-yesterday', cp_last_updated=datetime(2020, 1, 9))
+        self.index_form(domain.name, received_on=datetime(2020, 1, 8, 23, 59))
+
+        domains_filter = get_domains_to_update_es_filter()
+        results = DomainES().filter(domains_filter).fields(['_id', 'name']).run().hits
+
+        self.assertEqual(results, [])
+
+    def index_domain(self, name, active=True, cp_last_updated=None):
+        domain = create_domain(name, active)
```
[the remark this answers is not in the document]
Comment (gherceg) [reply corehq/apps/reports/tests/test_tasks.py:146]: See 160f597b9e2e0153e58cdee70c539f64873ee238

File: corehq/apps/reports/tasks.py [hunk 2]
```diff
@@ -142,8 +141,7 @@ def get_domains_to_update_es_filter():
         filters.OR(
             not_updated,
             more_than_a_week_ago,
-            filters.AND(less_than_a_week_ago,
-                        filters.term('name', domains_submitted_today))
+            filters.term('name', domains_submitted_today)
```
[the remark this answers is not in the document]
Comment (gherceg) [reply corehq/apps/reports/tasks.py]: Thanks for doing the digging I should have done! I did mean to verify if that line was valuable for an ES optimization that I was unaware of, but based on what you've linked to, that does not appear to be the case anymore.

**Comment by gherceg** [comment]
> I'm tempted to update the PR to still run calculated properties on inactive domains with submissions in the last day (or the most robust would be submissions since the last time calculated properties were updated). I'll look into confirming that today.

Noting I'm no longer tempted to do this :) The is_active field on domains represents if a domain has been confirmed yet, meaning there definitely aren't any meaningful metrics when is_active is false since it has not been used.

**Review by AmitPhulera — APPROVED** [review]

**Comment by gherceg** [comment]
I left this on staging for a week and confirmed that there aren't any inactive domains (`is_active=False`)  with `cp_last_updated` to a date on or after the date this change was deployed. Whereas compared to active domains, there are still domains being processed up to and including today. Merging 👍🏻

**Pull request merged**